homeassistant/components/hyperion/sensor.py

```diff
@@ -0,0 +1,212 @@
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+
+from . import (
+    get_hyperion_device_id,
+    get_hyperion_unique_id,
+    listen_for_instance_updates,
+)
+from .const import (
+    CONF_INSTANCE_CLIENTS,
+    DOMAIN,
+    HYPERION_MANUFACTURER_NAME,
+    HYPERION_MODEL_NAME,
+    SIGNAL_ENTITY_REMOVE,
+    TYPE_HYPERION_SENSOR_BASE,
+    TYPE_HYPERION_SENSOR_VISIBLE_PRIORITY,
+)
+
+SENSORS = [TYPE_HYPERION_SENSOR_VISIBLE_PRIORITY]
+
+
+def _sensor_unique_id(server_id: str, instance_num: int, suffix: str) -> str:
+    """Convert a sensor to a unique_id."""
+    return get_hyperion_unique_id(
+        server_id,
+        instance_num,
+        slugify(f"{TYPE_HYPERION_SENSOR_BASE}_{suffix}"),
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Hyperion platform from config entry."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    server_id = config_entry.unique_id
+
+    @callback
+    def instance_add(instance_num: int, instance_name: str) -> None:
+        """Add entities for a new Hyperion instance."""
+        assert server_id
+        sensors = []
+        sensors.append(
+            HyperionVisiblePrioritySensor(
+                server_id,
+                instance_num,
+                instance_name,
+                entry_data[CONF_INSTANCE_CLIENTS][instance_num],
+                SensorEntityDescription(
+                    key="visible_priority",
+                    translation_key="visible_priority",
+                    icon="mdi:lava-lamp",
+                    entity_registry_enabled_default=True,
+                ),
+            ),
+        )
+
+        async_add_entities(sensors)
+
+    @callback
+    def instance_remove(instance_num: int) -> None:
+        """Remove entities for an old Hyperion instance."""
+        assert server_id
+
+        for sensor in SENSORS:
+            async_dispatcher_send(
+                hass,
+                SIGNAL_ENTITY_REMOVE.format(
+                    _sensor_unique_id(server_id, instance_num, sensor),
+                ),
+            )
+
+    listen_for_instance_updates(hass, config_entry, instance_add, instance_remove)
+
+
+class HyperionSensor(SensorEntity):
+    """Sensor class."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    _client_callbacks: dict[str, Any] = {}
+
+    def __init__(
+        self,
+        server_id: str,
+        instance_num: int,
+        instance_name: str,
+        hyperion_client: client.HyperionClient,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = entity_description
+        self._device_id = get_hyperion_device_id(server_id, instance_num)
+        self._instance_name = instance_name
```

Comment:
`self._device_id` and `self._instance_name` only seem to be used locally in this method. Then we don't need to set the instance attributes. Just use the local variables.